odoo/addons/base/models/ir_qweb.py

```diff
@@ -613,16 +665,145 @@ def _render(self, template, values=None, **options):
             raise ValueError(f'values[{T_CALL_SLOT}] should be unset when call the _render method and only set into the template.')
 
         irQweb = self.with_context(**options)._prepare_environment(values)
+        irQweb = irQweb.with_context(
+            # List of generated and/or used functions, used for optimal performance
+            __qweb_loaded_functions={},
+            # List of codes generated during compilation. It is mainly used for debugging and displaying error messages.
+            __qweb_loaded_codes={},
+            # Reference to the last node being compiled. It is mainly used for debugging and displaying error messages.
+            _qweb_error_path_xml=[None, None],
+        )
 
         safe_eval.check_values(values)
 
-        template_functions, def_name = irQweb._compile(template)
-        render_template = template_functions[def_name]
-        rendering = render_template(irQweb, values)
-        result = ''.join(rendering)
+        root_values = values.copy()
+        values['__qweb_root_values'] = root_values['__qweb_root_values'] = root_values
+
+        result = ''.join(irQweb._render_iterall(template, None, values))
 
         return Markup(result)
 
+    def _render_iterall(self, view_ref, method, values):
```

Comment:
Nitpick. The type can help understand the comment.
```suggestion
    def _render_iterall(self, view_ref, method, values) -> Iterator[str | tuple[str, str, dict]]:
```